pkg/sql/create_table.go

```diff
@@ -873,7 +873,12 @@ func makeTableDescIfAs(
 func validateComputedColumnHasNoImpureFunctions(e tree.TypedExpr, colName tree.Name) error {
 	if fns := tree.ImpureFunctions(e); len(fns) != 0 {
 		var errMsg bytes.Buffer
-		errMsg.WriteString(fmt.Sprintf("computed column %s contains impure functions: ", colName))
+		errMsg.WriteString(
+			fmt.Sprintf(
+				"computed column %q contains impure functions: ",
```

Comment:
`&colName`

`(*tree.Name).String()` does the right thing

If in doubt use `tree.ErrString()` or `tree.NameString()`

Reply:
I don't fully understand what you're requesting here - `tree.ErrString` and `tree.NameString` operate on strings, rather than `tree.Name`s, no? And what is the difference between using a `*tree.Name` here and a `tree.Name`? Seems like if I do `%s` with `&colName` it doesn't auto-quote it.

Reply:
`".... %q ....", tree.ErrString(&colName)` is the reliably good way to go about doing this.

Using `%q` with a by-value `tree.Name` will auto-cast it to `string` and use that. Using `%q` with a by-pointer `*tree.Name` will call the `(*tree.Name) String()` method which does something else. We almost always want the latter.

Reply:
ErrString() is the way to say "do what I mean" without thinking about it too much

Reply:
Oh I see, I was confusing myself with the types somehow.